jcl/src/java.base/share/classes/openj9/internal/foreign/abi/LayoutStrPreprocessor.java

```diff
@@ -49,6 +51,60 @@
  */
 final class LayoutStrPreprocessor {
 
+	/*[IF JAVA_SPEC_VERSION <= 17]*/
+	private static final String osName = System.getProperty("os.name").toLowerCase(); //$NON-NLS-1$
+	private static final String arch = System.getProperty("os.arch").toLowerCase(); //$NON-NLS-1$
+	private static final String VARARGS_ATTR_NAME;
+
+	static {
+		/* Note: the attributes intended for the layout with variadic argument are defined in OpenJDK */
+		if ((arch.equals("amd64") || arch.equals("x86_64"))) { //$NON-NLS-1$ //$NON-NLS-2$
+			if (osName.startsWith("windows")) { //$NON-NLS-1$
+				VARARGS_ATTR_NAME = "abi/windows/varargs"; //$NON-NLS-1$
+			} else {
+				VARARGS_ATTR_NAME = "abi/sysv/varargs";; //$NON-NLS-1$
+			}
+		} else if (arch.equals("aarch64")) { //$NON-NLS-1$
+			if (osName.startsWith("mac")) { //$NON-NLS-1$
+				VARARGS_ATTR_NAME = "abi/aarch64/stack_varargs";; //$NON-NLS-1$
+			} else {
+				VARARGS_ATTR_NAME = "abi/sysv/varargs"; //$NON-NLS-1$;
+			}
+		} else if (arch.startsWith("ppc64")) { //$NON-NLS-1$
+			if (osName.startsWith("linux")) { //$NON-NLS-1$
+				VARARGS_ATTR_NAME = "abi/ppc64/sysv/varargs"; //$NON-NLS-1$
+			} else {
+				VARARGS_ATTR_NAME = "abi/ppc64/aix/varargs"; //$NON-NLS-1$
+			}
+		} else if (arch.equals("s390x") && osName.startsWith("linux")) { //$NON-NLS-1$ //$NON-NLS-2$
+			VARARGS_ATTR_NAME = "abi/s390x/sysv/varargs"; //$NON-NLS-1$
+		} else {
+			throw new ExceptionInInitializerError("Unsupported platform: " + arch + "_" + osName); //$NON-NLS-1$ //$NON-NLS-2$
```

Comment:
Do not throw `ExceptionInInitializerError` explicitly - the VM will wrap whatever exception you throw. I suggest simply using `Exception`.

Reply:
Replaced with `Exception` as suggested above.

Reply:
Changed to `InternalError` to pass the compilation.